### ISSUE TITLE
Inconsistent plan when having empty parameter blocks

### ISSUE THREAD

**Issue by Skaronator** [datadrivers/terraform-provider-nexus#81]
Hi, first of all: thank you for this provider and all your work!

I noticed a inconsistent plan when having empty parameter blocks vs. having no parameter block at all. My resource looks like this:

```hcl
resource "nexus_repository" "maven_central" {
  name   = "proxy-maven-central"
  type   = "proxy"
  format = "maven2"

  http_client {
    auto_block = true
    blocked    = false
  }

  negative_cache {
    enabled = true
    ttl     = 1440
  }

  proxy {
    remote_url = "https://repo1.maven.org/maven2/"
  }

  storage {
    blob_store_name                = "default"
    strict_content_type_validation = true
  }
}
```

When applying this I'll get the following plan and error:

```
Terraform will perform the following actions:

  # nexus_repository.maven_central will be created
  + resource "nexus_repository" "maven_central" {
      + format = "maven2"
      + id     = (known after apply)
      + name   = "proxy-maven-central"
      + online = true
      + type   = "proxy"

      + http_client {
          + auto_block = true
          + blocked    = false
        }

      + negative_cache {
          + enabled = true
          + ttl     = 1440
        }

      + proxy {
          + content_max_age  = 1440
          + metadata_max_age = 1440
          + remote_url       = "https://repo1.maven.org/maven2/"
        }

      + storage {
          + blob_store_name                = "default"
          + strict_content_type_validation = true
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

nexus_repository.maven_central: Creating...

Error: could not create repository 'proxy-maven-central': HTTP: 400, [ {
  "id" : "PARAMETER maven",
  "message" : "must not be null"
} ]

  on repositories-cache.tf line 37, in resource "nexus_repository" "maven_central":
  37: resource "nexus_repository" "maven_central" {
```

When adding a empty `maven {}` parameter block to my resource like so:

```diff
resource "nexus_repository" "maven_central" {
  name   = "proxy-maven-central"
  type   = "proxy"
  format = "maven2"

+ maven {}

  http_client {
    auto_block = true
    blocked    = false
  }

  negative_cache {
    enabled = true
    ttl     = 1440
  }

  proxy {
    remote_url = "https://repo1.maven.org/maven2/"
  }

  storage {
    blob_store_name                = "default"
    strict_content_type_validation = true
  }
}
```

My plan changes to this:

```
# nexus_repository.maven_central will be created
  + resource "nexus_repository" "maven_central" {
      + format = "maven2"
      + id     = (known after apply)
      + name   = "proxy-maven-central"
      + online = true
      + type   = "proxy"

      + http_client {
          + auto_block = true
          + blocked    = false
        }

      + maven {
          + layout_policy  = "PERMISSIVE"
          + version_policy = "RELEASE"
        }

      + negative_cache {
          + enabled = true
          + ttl     = 1440
        }

      + proxy {
          + content_max_age  = 1440
          + metadata_max_age = 1440
          + remote_url       = "https://repo1.maven.org/maven2/"
        }

      + storage {
          + blob_store_name                = "default"
          + strict_content_type_validation = true
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

As you can see above my Terraform Plan changes. There is now a `maven` block with some default parameters!

```
      + maven {
          + layout_policy  = "PERMISSIVE"
          + version_policy = "RELEASE"
        }
```

This feels like a bug, doesn't it? Either mark this maven block as required or always pass the default values for this block.

_____
Versions:
Terraform v0.13.0
+ provider registry.terraform.io/datadrivers/nexus v1.10.2

**Comment by Skaronator** [comment]
I think this would be fixed by the proposal in #15?

**Comment by anmoel** [comment]
Hi @Skaronator,

please use the new repository resources.

I will close this issue. Please open a new ticket if a new resource is not working correctly.
